library/ssl_tls.c

```diff
@@ -7231,8 +7239,8 @@ int mbedtls_ssl_config_defaults( mbedtls_ssl_config *conf,
 
 #if defined(MBEDTLS_SSL_RENEGOTIATION)
     conf->renego_max_records = MBEDTLS_SSL_RENEGO_MAX_RECORDS_DEFAULT;
-    memset( conf->renego_period, 0xFF, 7 );
-    conf->renego_period[7] = 0x00;
+    memset( conf->renego_period,     0xFF, 6 );
+    memset( conf->renego_period + 6, 0x00, 2 );
```

Comment:
Ok, I see what the confusion is. The problem is that in DTLS the renegotiation period is 6 bytes but we are comparing 8. Internally, the SSL module has three pointers that keep track of the record counter (same as renegotiation period):
* `renego_period`: an 8 byte array that keeps the number of exchanges before renegotiation is performed.
* `in_ctr`: record counter of incoming messages.
* `out_ctr`: record counter of outgoing messages.
There are two cases:
* Using TLS:
    * Strictly speaking, TLS does not send record counters, but our SSL module maintains an 8 byte record counter that is compared to `renego_period` to trigger renegotiation routines. This works just fine.
* Using DTLS:
    * The DTLS input and output records contain a 6 byte record counter and a 2 byte epoch value.

The problem that this PR is trying to solve is that our SSL module believes that the record counter for both TLS and DTLS is 8 bytes long. As I explained, this is not the case for DTLS, so when comparing `renego_period` with say `in_ctr` our SSL module ccompares 6 bytes of record counter AND 2 bytes that in the input or output messages correspond to the epoch value. This is clearly wrong and causes renegotiation to be trigger when it should not if the value in`renego_period` is greater than `2^48 - 1`. To fix this comparison I modified the function `ssl_check_ctr_renegotiate()`.

However, there is another problem. The documentation is misleading because it says that the value of the `renego_period` is 8 bytes and the functions `mbedtls_ssl_config_defaults()` and `mbedtls_ssl_conf_renegotiation_period()` treat it in this way. So instead of changing types for `renego_period` what I have done is to adjust the documentation to say that the maximum value for `renego_period` is 2^48 -1 for DTLS and 2^64 -1 for TLS. To make it consistent I also modified the start value to be 2^48 - 1 instead of 2^64 - 1, which is the change that is in lines 7242 and 7243, even though strictly speaking this is not necessary.